### DISCUSSION TITLE
请问系统环境变量怎么配置？手册没有找到

### DISCUSSION THREAD

**Discussion by lucas-wu-wakakak** [xmake-io/xmake#2986] [Q&A]
$ xmake show -l envs
XMAKE_RAMDIR            Set the ramdisk directory.
                        <empty>
XMAKE_GLOBALDIR         Set the global config directory of xmake.
                        /Users/ruki/.xmake
XMAKE_ROOT              Allow xmake to run under root.
                        <empty>
XMAKE_COLORTERM         Set the color terminal environment.
                        <empty>
XMAKE_PKG_INSTALLDIR    Set the install directory of packages.
                        <empty>
XMAKE_TMPDIR            Set the temporary directory.
                        /var/folders/vn/ppcrrcm911v8b4510klg9xw80000gn/T/.xmake501/211104
XMAKE_PKG_CACHEDIR      Set the cache directory of packages.
                        <empty>
XMAKE_PROGRAM_DIR       Set the program scripts directory of xmake.
                        /Users/ruki/.local/share/xmake
XMAKE_PROFILE           Start profiler, e.g. perf, trace.
                        <empty>
XMAKE_RCFILES           Set the runtime configuration files.

XMAKE_CONFIGDIR         Set the local config directory of project.
                        /Users/ruki/projects/personal/xmake-docs/.xmake/macosx/x86_64
XMAKE_LOGFILE           Set the log output file path.
                        <empty>

XMAKE_CONFIGDIR 和XMAKE_RAMDIR请问怎么配置呀？有具体的命令行提示么？

**Comment by waruqi** [comment]
https://xmake.io/#/zh-cn/guide/configuration?id=%e7%8e%af%e5%a2%83%e5%8f%98%e9%87%8f

**Reply by lucas-wu-wakakak** [reply]
[XMAKE_CONFIGDIR](https://xmake.io/#/zh-cn/guide/configuration?id=xmake_configdir)
设置本地工程配置目录
每个项目的本地编译配置，默认会存储在当前项目根目录的 .xmake 路径下，然后根据不同的平台，架构区分，例如：

.xmake/macosx/x86_64
Copy to clipboardErrorCopied
我们如果不想存储在项目根目录，也可以自己设置到其他路径，比如 build 目录下等等。

[XMAKE_CONFIGDIR](https://xmake.io/#/guide/configuration?id=xmake_configdir)
Set local project configuration directory
The local compilation configuration of each project will be stored in the .xmake path in the root directory of the current project by default, and then differentiated according to different platforms and architectures, for example:

.xmake/macosx/x86_64
Copy to clipboardErrorCopied
If we don't want to store it in the root directory of the project, we can also set it to other paths ourselves, such as the build directory and so on.


命令是xmake f --XMAKE_CONFIGDIR=c:/temp ? 
或者xmake f --XMAKE_CONFIGDIR c:/temp ? 
还是 xmake g --XMAKE_CONFIGDIR=c:/temp ? 
还是在代码里export XMAKE_CONFIGDIR=c:/temp ?

**Reply by waruqi** [reply]
google 下 win 环境变量设置

**Reply by lucas-wu-wakakak** [reply]
麻烦补充一句在pc环境中设置，真的容易蒙圈，我以为是xmake里面类似于builddir这样的环境变量。

**Reply by waruqi** [reply]
都说了是 ”环境变量“ 不是 “变量”， 还能是哪个。。

**Reply by lucas-wu-wakakak** [reply]
大佬，你是写这个软件的人，当然清楚加了环境是为了区分一般变量。但是有的使用者有时候不一定会一下就能get到区别。。。。。
